### DISCUSSION TITLE
Advice for adding NextJS support to Expo project

### DISCUSSION THREAD

**Discussion by JayPerfetto** [infinitered/ignite#1757] [Q&A]
Does anyone have any experience using NextJS with Ignite + Expo? I am not able to get the boilerplate to compile when I follow the Expo instructions to add NextJS to an existing expo project, running yarn next dev

Thanks for any pointers!

**Comment by jamonholmgren** [comment]
Could you provide some more information on what compilation errors you're running into?

**Reply by JayPerfetto** [reply]
<img width="595" alt="Screen Shot 2021-08-06 at 3 53 42 PM" src="https://user-images.githubusercontent.com/34031697/128578251-5a9b9ad4-7467-40f4-bd1f-49a5a0c1a459.png">
Perhaps I am just confused by the instructions here, and maybe this isn't possible at all using Ignite's boilerplate - but - I am going through these instructions after creating the sample PizzaApp passing --expo, and running `yarn next dev`

I don't see any compilation errors, but going to localhost:3000 goes to an empty black page and never actually loads. I am still able to run 'expo start' and that launches the normal Expo server at localhost:1900X.

I guess I am just confused if I'm missing a step thats not included in the docs to make the Ignite app serve over localhost:3000 when running yarn net dev....This is all I see in my terminal when I run that:
<img width="878" alt="Screen Shot 2021-08-06 at 4 09 21 PM" src="https://user-images.githubusercontent.com/34031697/128579020-04900213-4595-4f67-a4ce-ec0d7d0e62cc.png">

**Reply by JayPerfetto** [reply]
Here is a comparison with my results from:
<img width="1439" alt="Screen Shot 2021-08-06 at 4 14 46 PM" src="https://user-images.githubusercontent.com/34031697/128579306-a2f64fb6-fb6b-4eb5-b2a3-a9b3360cf8e9.png">

`npx create-react-native-app -t with-nextjs
`

**Reply by JayPerfetto** [reply]
You can see in the attempt using Ignite, I never see the "next-transpile-modules" message, nor any subsequent logging

**Reply by jamonholmgren** [reply]
@JayPerfetto I didn't see your response until now. Did you get this figured out yet?

**Reply by JayPerfetto** [reply]
Not really - ultimately just elected to move forward with a separate dedicated NextJS project, ill share code where I can but it didn't make sense for my Org to try to keep forcing the "write-once" philosophy where it was ultimately just slowing us down and reducing quality 

Thanks for the follow up tho!

**Comment by QasidLabeed** [comment]
Have anyone tried React Native CLI with NextJs using Ignite. 
Any explanation in this regard would be great. Thanks

**Reply by SaltMango** [reply]
> Have anyone tried React Native CLI with NextJs using Ignite. Any explanation in this regard would be great. Thanks

Were you able to do it?
